api/src/opentrons/protocol_runner/legacy_command_mapper.py

```diff
@@ -192,78 +229,223 @@ def _build_initial_command(
         now: datetime,
     ) -> pe_commands.Command:
         engine_command: pe_commands.Command
-        well: LegacyWell
-        if (
-            command["name"] == legacy_command_types.PICK_UP_TIP
-            and "instrument" in command["payload"]
-            and "location" in command["payload"]
-            and isinstance(command["payload"]["location"], LegacyWell)
+        if command["name"] == legacy_command_types.PICK_UP_TIP:
+            engine_command = self._build_pick_up_tip_command(
+                command=command, command_id=command_id, now=now
+            )
+        elif command["name"] == legacy_command_types.DROP_TIP:
+            engine_command = self._build_drop_tip_command(
+                command=command, command_id=command_id, now=now
+            )
+
+        elif (
+            command["name"] == legacy_command_types.ASPIRATE
+            or command["name"] == legacy_command_types.DISPENSE
         ):
-            pipette: LegacyPipetteContext = command["payload"]["instrument"]
-            _well = command["payload"].get("location")
-            if isinstance(_well, LegacyWell):
-                well = _well
-            else:
-                raise Exception("Unknown pick_up_tip location.")
+            engine_command = self._build_liquid_handling_commands(
+                command=command, command_id=command_id, now=now
+            )
+        elif command["name"] == legacy_command_types.BLOW_OUT:
+            engine_command = self._build_blow_out_command(
+                command=command, command_id=command_id, now=now
+            )
+        elif command["name"] == legacy_command_types.PAUSE:
+            engine_command = pe_commands.WaitForResume.construct(
+                id=command_id,
+                key=command_id,
+                status=pe_commands.CommandStatus.RUNNING,
+                createdAt=now,
+                startedAt=now,
+                params=pe_commands.WaitForResumeParams.construct(
+                    message=command["payload"]["userMessage"],
+                ),
+            )
+        else:
+            engine_command = pe_commands.Custom.construct(
+                id=command_id,
+                key=command_id,
+                status=pe_commands.CommandStatus.RUNNING,
+                createdAt=now,
+                startedAt=now,
+                params=LegacyCommandParams.construct(
+                    legacyCommandType=command["name"],
+                    legacyCommandText=command["payload"]["text"],
+                ),
+            )
+
+        return engine_command
+
+    def _build_drop_tip_command(
+        self,
+        command: legacy_command_types.DropTipMessage,
+        command_id: str,
+        now: datetime,
+    ) -> pe_commands.Command:
+        pipette: LegacyPipetteContext = command["payload"]["instrument"]
+        location = command["payload"]["location"]
+        if location.labware.is_well:
+            well = location.labware.as_well()
```

Comment:
Can we change this to an assert? The Protocol API ensures that only a well `Location` will ever make it here, so the `if` is a little misleading, because the `else` would be a violation of our internal guarantees 

```suggestion
        assert location.labware.is_well
        well = location.labware.as_well()
```

You can tell this `if` is bad because it's the only place where the `well` variable is assigned a value, but it's used later in the function outside of the `if`